File: rust/src/models/cancel_subscription_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::subscription::Subscription;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CancelSubscriptionResponse {
    pub subscription: Subscription,
}

impl CancelSubscriptionResponse {
    pub fn new(subscription: Subscription) -> Self {
        Self { subscription }
    }
}

File: typescript/src/models/cancelSubscriptionResponse.ts
// this file is @generated
import { type Subscription, SubscriptionSerializer } from "./subscription";

export interface CancelSubscriptionResponse {
  subscription: Subscription;
}

export const CancelSubscriptionResponseSerializer = {
  _fromJsonObject(object: any): CancelSubscriptionResponse {
    return {
      subscription: SubscriptionSerializer._fromJsonObject(object["subscription"]),
    };
  },

  _toJsonObject(self: CancelSubscriptionResponse): any {
    return {
      subscription: SubscriptionSerializer._toJsonObject(self.subscription),
    };
  },
};
